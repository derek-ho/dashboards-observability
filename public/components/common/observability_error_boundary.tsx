/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiErrorBoundary, EuiLink, EuiSpacer, EuiText } from '@elastic/eui';
import React from 'react';
import {
  OBSERVABILITY_GITHUB_ISSUE_URL,
  OPENSEARCH_OBSERVABILITY_FORUM_URL,
} from '../../../common/constants/shared';

export function ObservabilityErrorBoundary(props: { children: React.ReactNode }) {
  return (
    <div>
      <EuiErrorBoundary children={props.children} />
      <EuiSpacer size="l" />
      <EuiText size="s" color="subdued">
      An unexpected error occurred. Please refresh the page, and if the issue persists, report it by creating an issue on{' '}
        <EuiLink external={true} href={OBSERVABILITY_GITHUB_ISSUE_URL} target="blank">
        GitHub
        </EuiLink>{' '}
        or posting on the{' '}
        <EuiLink external={true} href={OPENSEARCH_OBSERVABILITY_FORUM_URL} target="blank">
        community forum
        </EuiLink>
      </EuiText>
    </div>
  );
}
