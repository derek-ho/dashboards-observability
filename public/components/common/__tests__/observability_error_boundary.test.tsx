/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { ObservabilityErrorBoundary } from '../observability_error_boundary';

describe('Traces table component', () => {
  configure({ adapter: new Adapter() });

  it('renders empty traces table message', () => {
    const BadComponent = () => {
      throw new Error('Testing Observability Error Boundary');
    };
    const errorScreen = mount(
      <ObservabilityErrorBoundary>
        <BadComponent />
      </ObservabilityErrorBoundary>
    );
    expect(errorScreen).toMatchSnapshot();
  });
});
