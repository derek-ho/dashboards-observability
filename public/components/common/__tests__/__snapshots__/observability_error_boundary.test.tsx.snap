// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Traces table component renders empty traces table message 1`] = `
<ObservabilityErrorBoundary>
  <div>
    <EuiErrorBoundary>
      <div
        className="euiErrorBoundary"
      >
        <div
          className="euiErrorBoundary__text"
        >
          <EuiText
            size="xs"
          >
            <div
              className="euiText euiText--extraSmall"
            >
              <h1>
                Error
              </h1>
              <pre
                className="euiErrorBoundary__stack"
              >
                <p>
                  Error: Testing Observability Error Boundary
    at Component (/Users/dxho/Desktop/OpenSearch-Dashboards/plugins/dashboards-observability/public/components/common/__tests__/observability_error_boundary.test.tsx:16:15)
    at renderWithHooks (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:14803:18)
    at mountIndeterminateComponent (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:17482:13)
    at apply (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:18596:16)
    at HTMLUnknownElement.callCallback (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:188:14)
    at HTMLUnknownElement.callTheUserObjectsOperation (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jsdom/lib/jsdom/living/generated/EventListener.js:26:30)
    at innerInvokeEventListeners (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:338:25)
    at invokeEventListeners (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:274:3)
    at HTMLUnknownElementImpl._dispatch (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:221:9)
    at HTMLUnknownElementImpl.dispatchEvent (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:94:17)
    at HTMLUnknownElement.dispatchEvent (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:231:34)
    at Object.apply (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:237:16)
    at invokeGuardedCallback (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:292:31)
    at beginWork$1 (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:23203:7)
    at performUnitOfWork (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:22157:12)
    at workLoopSync (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:22130:22)
    at performSyncWorkOnRoot (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:21756:9)
    at scheduleWork (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:21188:7)
    at updateContainer (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:24373:3)
    at fn (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:24758:7)
    at unbatchedUpdates (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:21903:12)
    at legacyRenderSubtreeIntoContainer (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:24757:5)
    at Object.render (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:24840:10)
    at fn (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:488:26)
    at fn (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:405:37)
    at batchedUpdates (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom.development.js:21856:12)
    at Object.act (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/react-dom/cjs/react-dom-test-utils.development.js:929:14)
    at wrapAct (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:405:13)
    at Object.render (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:474:16)
    at new ReactWrapper (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/enzyme/src/ReactWrapper.js:115:16)
    at mount (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/enzyme/src/mount.js:10:10)
    at Object.&lt;anonymous&gt; (/Users/dxho/Desktop/OpenSearch-Dashboards/plugins/dashboards-observability/public/components/common/__tests__/observability_error_boundary.test.tsx:18:25)
    at Promise.then.completed (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jest-circus/build/utils.js:391:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jest-circus/build/utils.js:316:10)
    at _callCircusTest (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jest-circus/build/run.js:218:40)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at _runTest (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jest-circus/build/run.js:155:3)
    at _runTestsForDescribeBlock (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jest-circus/build/run.js:66:9)
    at _runTestsForDescribeBlock (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jest-circus/build/run.js:60:9)
    at run (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jest-circus/build/run.js:25:3)
    at runAndTransformResultsToJestFormat (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)
    at jestAdapter (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:81:19)
    at runTestInternal (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jest-runner/build/runTest.js:389:16)
    at runTest (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jest-runner/build/runTest.js:475:34)
    at Object.worker (/Users/dxho/Desktop/OpenSearch-Dashboards/node_modules/jest-runner/build/testWorker.js:133:12)
                </p>
              </pre>
            </div>
          </EuiText>
        </div>
      </div>
    </EuiErrorBoundary>
    <EuiSpacer
      size="l"
    >
      <div
        className="euiSpacer euiSpacer--l"
      />
    </EuiSpacer>
    <EuiText
      color="subdued"
      size="s"
    >
      <div
        className="euiText euiText--small"
      >
        <EuiTextColor
          color="subdued"
          component="div"
        >
          <div
            className="euiTextColor euiTextColor--subdued"
          >
            An unexpected error occurred. Please refresh the page, and if the issue persists, report it by creating an issue on
             
            <EuiLink
              external={true}
              href="https://github.com/opensearch-project/dashboards-observability/issues"
              target="blank"
            >
              <a
                className="euiLink euiLink--primary"
                href="https://github.com/opensearch-project/dashboards-observability/issues"
                rel="noreferrer"
                target="blank"
              >
                GitHub
                <EuiIcon
                  aria-label="External link"
                  className="euiLink__externalIcon"
                  size="s"
                  type="popout"
                >
                  <EuiIconEmpty
                    aria-hidden={true}
                    aria-label="External link"
                    className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                    focusable="false"
                    role="img"
                    style={null}
                  >
                    <svg
                      aria-hidden={true}
                      aria-label="External link"
                      className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                      focusable="false"
                      height={16}
                      role="img"
                      style={null}
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </EuiIconEmpty>
                </EuiIcon>
              </a>
            </EuiLink>
             
            or posting on the
             
            <EuiLink
              external={true}
              href="https://forum.opensearch.org/tag/observability"
              target="blank"
            >
              <a
                className="euiLink euiLink--primary"
                href="https://forum.opensearch.org/tag/observability"
                rel="noreferrer"
                target="blank"
              >
                community forum
                <EuiIcon
                  aria-label="External link"
                  className="euiLink__externalIcon"
                  size="s"
                  type="popout"
                >
                  <EuiIconEmpty
                    aria-hidden={true}
                    aria-label="External link"
                    className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                    focusable="false"
                    role="img"
                    style={null}
                  >
                    <svg
                      aria-hidden={true}
                      aria-label="External link"
                      className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                      focusable="false"
                      height={16}
                      role="img"
                      style={null}
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </EuiIconEmpty>
                </EuiIcon>
              </a>
            </EuiLink>
          </div>
        </EuiTextColor>
      </div>
    </EuiText>
  </div>
</ObservabilityErrorBoundary>
`;
